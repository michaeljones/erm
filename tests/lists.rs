extern crate codespan_reporting;
extern crate erm;
extern crate im;
extern crate logos;
extern crate unindent;

mod common;

mod lists {

    use common::eval;

    #[test]
    fn empty_list_literal() {
        let src = r#"
        module Main exposing (..)
        main args =
          String.join "," []
        "#;
        let result = eval(src, None);
        insta::assert_snapshot!(result);
    }

    #[test]
    fn list_literal_with_strings() {
        let src = r#"
        module Main exposing (..)
        main args =
          String.join "," ["Hello", " World"]
        "#;
        let result = eval(src, None);
        insta::assert_snapshot!(result);
    }

    #[test]
    fn list_literal_with_inconsistent_types() {
        let src = r#"
        module Main exposing (..)
        main args =
          String.join "," ["Hello", 1]
        "#;
        let result = eval(src, None);
        insta::assert_snapshot!(result);
    }

    #[test]
    fn list_sum() {
        let src = r#"
        module Main exposing (..)
        main args =
          String.fromInt (List.sum [1, 2])
        "#;
        let result = eval(src, None);
        insta::assert_snapshot!(result);
    }
}
