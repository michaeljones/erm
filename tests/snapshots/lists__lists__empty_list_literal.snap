---
source: tests/lists.rs
expression: result

---

